services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java

```diff
@@ -101,17 +101,16 @@ public static final boolean canEditChallenge(List<Long> managersId, String space
     return isChallengeOwner && isSpaceManager;
   }
 
-  public static final boolean canAnnounce(String id) {
+  public static final boolean canAnnounce(String spaceId, String username) {
     SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
```

Comment:
It would be better to not retrieve services statically but add depending services as parameter inside the method signature to make sure that the caller Service has a dependency on it. (This should be used only by components that aren't managed by IOC)
( This change is optional, and just for best practices knowledge share )